packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/TypeProvider.cs

```diff
@@ -361,10 +361,42 @@ private bool ShouldGenerate(PropertyProvider property)
                 }
             }
 
-            if (customPropertyReplacements.TryGetValue(property.Name, out PropertyProvider? customProp))
+            var customPropertySerializationNames = new Dictionary<string, string>();
+            foreach (var attribute in GetCodeGenSerializationAttributes())
+            {
+                if (CodeGenAttributes.TryGetCodeGenSerializationAttributeValue(
+                    attribute,
+                    out var propertyName,
+                    out IReadOnlyList<string>? serializationNames,
+                    out _,
+                    out _,
+                    out _) && serializationNames?.Count > 0)
+                {
+                    customPropertySerializationNames.Add(propertyName, serializationNames[0]);
+                }
+            }
+
+            var customPropReplacementExists = customPropertyReplacements.TryGetValue(property.Name, out PropertyProvider? customProp);
+
+            if (property.WireInfo != null)
+            {
+                // replace original property serialization name. Otherwise try replacing custom property serialization name
+                if (customPropertySerializationNames.TryGetValue(property.Name, out var serializedName) && serializedName != null)
+                {
+                    property.WireInfo.SerializedName = serializedName;
+                }
+                else if (customPropReplacementExists && customProp != null
+                    && customPropertySerializationNames.TryGetValue(customProp.Name, out serializedName) && serializedName != null)
+                {
+                    customProp.WireInfo = property.WireInfo;
```

Comment:
Rather than setting the custom property WireInfo in two places, can we just store the serializedName here and then set that below where we set the wireInfo?